diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3525d3fc..833c906b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,8 +51,9 @@ include(Utilities)
 include(Macros)
 include(CMakeDependentOption)
 include(CheckCCompilerFlag)
+include(CheckCXXSymbolExists)
 
-project(CURL C)
+project(CURL C CXX)
 
 file(STRINGS ${CURL_SOURCE_DIR}/include/curl/curlver.h CURL_VERSION_H_CONTENTS REGEX "#define LIBCURL_VERSION( |_NUM )")
 string(REGEX MATCH "#define LIBCURL_VERSION \"[^\"]*"
@@ -628,7 +629,7 @@ macro(openssl_check_symbol_exists SYMBOL FILES VARIABLE EXTRA_LIBS)
   if(NOT "${EXTRA_LIBS}" STREQUAL "")
     list(APPEND CMAKE_REQUIRED_LIBRARIES "${EXTRA_LIBS}")
   endif()
-  check_symbol_exists("${SYMBOL}" "${FILES}" "${VARIABLE}")
+  check_cxx_symbol_exists("${SYMBOL}" "${FILES}" "${VARIABLE}")
   cmake_pop_check_state()
 endmacro()
 
